npm/ng-packs/packages/core/src/lib/services/routes.service.ts

```diff
@@ -171,15 +183,17 @@ export abstract class AbstractTreeService<T extends { [key: string | number | sy
   search(params: Partial<T>, tree = this.tree): TreeNode<T> | null {
     const searchKeys = Object.keys(params) as Array<keyof Partial<T>>;
 
-    return tree.reduce<TreeNode<T> | null>(
-      (acc, node) =>
-        acc
-          ? acc
-          : searchKeys.every(key => node[key] === params[key])
-            ? node
-            : this.search(params, node.children),
-      null,
-    );
+    return tree.reduce<TreeNode<T> | null>((acc, node) => {
+      if (acc) {
+        return acc;
+      }
+
+      if (searchKeys.every(key => node.item[key] === params[key])) {
```

Comment:
`node.item[key]`is broke the tests. The old version `node[key]` is working. I'm including this change in my latest commit.